capture/crawler_valor.py

```diff
@@ -0,0 +1,203 @@
+    :return: news published datetime
+    :rtype: string
+	"""
+	try:
+		time_tag = soup.find(id="content-area").find_all('span', class_='date submitted')[0].text
+	except IndexError:
+		logger.error('wrong time tag')
+		return None
+	if time_tag is None:
+		return None
+	else:
+		try:
+			published_time = datetime.datetime.strptime( time_tag.encode('utf8') , '%d/%m/%Y às %Hh%M')
+		except ValueError:
+			logger.error('wrong date extraction')
+			return None
+		return published_time
+
+def extract_title(article):
+    """
+	Extract the news title.
+    """
+
+    try:
+        title = article.title
+    except Exception as ex:
+        template = "An exception of type {0} occured during extraction of news title. Arguments:\n{1!r}"
+        message = template.format(type(ex).__name__, ex.args)
+        logger.exception(message)
+        return None
+    return title
+
+def extract_content(article):
+    """
+	Extract relevant information about news page
+    """
+
+    try:
+        body_content = article.cleaned_text
+    except Exception as ex:
+        template = "An exception of type {0} occured during extraction of news content. Arguments:\n{1!r}"
+        message = template.format(type(ex).__name__, ex.args)
+        logger.exception(message)
+        return None
+    return body_content
+
+def detect_language(text):
+    """
+	Detect the language of text using chromium_compact_language_detector
+    :param text: text to be analyzed
+    :return: {"name": portuguese, "pt"}
+    """
+    name, code, isReliable, textBytesFound, details = cld.detect(text.encode('utf8'))
+    return {"name": name, "code": code}
+
+def compress_content(html):
+    """
+    Compresses and encodes html content so that it can be BSON encoded an store in mongodb
+    :param html: original html document
+    :return: compressed an b64 encoded document
+    """
+    pickled = CP.dumps(html, CP.HIGHEST_PROTOCOL)
+    squished = zlib.compress(pickled)
+    encoded = bson.Binary(squished)  # b64.urlsafe_b64encode(squished)
+    return encoded
+
+def decompress_content(compressed_html):
+    """
+    Decompress data compressed by `compress_content`
+    :param compressed_html: compressed html document
+    :return: original html
+    """
+    # unencoded = b64.urlsafe_b64decode(str(compressed_html))
+    decompressed = zlib.decompress(compressed_html)
+    orig_html = CP.loads(decompressed)
+    return orig_html
+
+
+def download_article(url):
+	"""
+	Download the html content of a news page
+    :param url: news page's url
+    :type url: string
+    :return: news page's content
+    :rtype: requests.models.Response	
+	"""
+	article = {
+        'link': url,
+        'source': 'crawler_Valor',
+    }
+	logger.info("Downloading article: %s", url)
+	try:
+		response = requests.get(url, timeout=30)
+	except ConnectionError:
+		logger.error("Failed to fetch %s", url)
+		return
+	except Timeout:
+		logger.error("Timed out while fetching %s", url)
```

Comment:
I'd use new style string formatting here, just to keep it consistent and make the move to python3 easier later on.